src/core/utils.ts

```diff
@@ -162,8 +204,8 @@ export function parseFilterArgs<
   arg3?: TOptions
 ): [TFilters, TOptions | undefined] {
   return (isQueryKey(arg1)
-    ? [{ ...arg2, queryKey: arg1 }, arg3]
-    : [arg1 || {}, arg2]) as [TFilters, TOptions]
+    ? [{ ...getQueryFilters(arg2), queryKey: arg1 }, arg3]
+    : [getQueryFilters(arg1) || {}, arg2]) as [TFilters, TOptions]
```

Comment:
why is this change necessary please?

Reply:
At the moment, any keys outside of the QueryFilters ones still get passed.
Some types (functions for example) are not compatible with being broadcast, resulting in errors.
`getQueryFilters` narrows the type to only include `QueryFilters` without extra keys.

Reply:
Why would it have extra keys? predicate is also a function, so how does it work with that valid key? If this change is specifically for the broadcast plugin, I think it would be better if it lived there rather than in the core

Reply:
I don't remember exactly, maybe a type solution would be more appropriate?
Good point, I've never used predicate I think, this might block the whole thing though :/